lib/inventory_refresh/save_collection/saver/base.rb

```diff
@@ -234,12 +232,12 @@ def assert_distinct_relation(primary_key_value)
           # Change the InventoryCollection's :association or :arel parameter to return distinct results. The :through
           # relations can return the same record multiple times. We don't want to do SELECT DISTINCT by default, since
           # it can be very slow.
-          unless inventory_collection.assert_graph_integrity
+          if inventory_collection.assert_graph_integrity
```

Comment:
What a strangely named method...it's an assert_graph_integrity would imply it would return true if the graph has good integrity, but instead it returns false. 😖